src/main/scala/faces/utils/Helpers.scala

```diff
@@ -287,15 +290,398 @@ case class Helpers(cfg: FacesSettings)(implicit rnd: Random) {
     withLight
   }
 
-  def writeImg(img: PixelImage[RGBA], id: Int, n: Int, postfix: String): Unit = {
+  lazy val loadTextures: IndexedSeq[File] = {
+    // search all texture files
+    new File(inTexturesPath).listFiles.filter(_.getName.endsWith(".png")).toIndexedSeq
+  }
+
+  def writeImg(img: PixelImage[RGBA], id: Int, n: Int, postfix: String, mask: PixelImage[RGBA]): PixelImage[RGBA] = {
```

Comment:
please have a writeImg without mask that generates a mask and calles writeImg with mask (to avoid your trick to call it with twice the image)